interfaces/mount/spec.go

```diff
@@ -39,10 +46,103 @@ func (spec *Specification) AddMountEntry(e Entry) error {
 	return nil
 }
 
+// resolveSpecialVariable resolves $SNAP, $SNAP_DATA and $SNAP_COMMON.
+func resolveSpecialVariable(path string, snapInfo *snap.Info) string {
+	return os.Expand(path, func(v string) string {
+		switch v {
+		case "SNAP":
+			// NOTE: We use dirs.CoreSnapMountDir here as the path used will be always
+			// inside the mount namespace snap-confine creates and there we will
+			// always have a /snap directory available regardless if the system
+			// we're running on supports this or not.
+			return filepath.Join(dirs.CoreSnapMountDir, snapInfo.Name(), snapInfo.Revision.String())
+		case "SNAP_DATA":
+			return snapInfo.DataDir()
+		case "SNAP_COMMON":
+			snapInfo.CommonDataDir()
+		}
+		return ""
+	})
+}
+
+func mountEntryFromLayout(layout *snap.Layout) Entry {
+	var entry Entry
+
+	mountPoint := resolveSpecialVariable(layout.Path, layout.Snap)
+	entry.Dir = mountPoint
+
+	if layout.Bind != "" {
+		mountSource := resolveSpecialVariable(layout.Bind, layout.Snap)
+		// XXX: what about ro mounts?
+		// XXX: what about file mounts, those need x-snapd.kind=file to create correctly?
+		entry.Options = []string{"bind", "rw"}
+		entry.Name = mountSource
+	}
+
+	if layout.Type == "tmpfs" {
+		entry.Type = "tmpfs"
+		entry.Name = "tmpfs"
+	}
+
+	if layout.Symlink != "" {
+		oldname := resolveSpecialVariable(layout.Symlink, layout.Snap)
+		entry.Options = []string{"x-snapd.kind=symlink", fmt.Sprintf("x-snapd.symlink=%s", oldname)}
+	}
+
+	var uid int
+	switch layout.User {
+	case "root", "":
+		uid = 0
+	case "nobody":
+		// The user "nobody" has a fixed value in the Ubuntu core snap.
+		// TODO: load this from an attribute in other bases or require the same ID.
+		uid = 65534
+	}
+	if uid != 0 {
+		entry.Options = append(entry.Options, fmt.Sprintf("x-snapd.user=%d", uid))
+	}
+
+	var gid int
+	switch layout.Group {
+	case "root", "":
```

Comment:
In the light of your answer to one of @mvo5 's questions, could you please add  a comment explaining that only "root" and "nobody" are expected here.

Reply:
Sure